code/datums/components/autofill.dm

```diff
@@ -0,0 +1,89 @@
+/// Automatically fills something with reagents... and possible leaks it
+TYPEINFO(/datum/component/auto_reagent)
+	initialization_args = list(
+		ARG_INFO("reagent_id", DATA_INPUT_TEXT, "Reagent to overflow/produce"),
+		ARG_INFO("units", DATA_INPUT_NUM, "Units to produce per cycle", 10),
+		ARG_INFO("overflowing", DATA_INPUT_BOOL, "Should overflow?", FALSE),
+	)
+/datum/component/auto_reagent
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Reagent ID to produce
+	var/reagent_id = null
+	/// Number of reagents to produce per item process
+	var/units = 10
+	var/overflowing = FALSE
+	var/_added_to_items_processing = FALSE
+
+	Initialize(reagent_id=null, units=10, overflowing=FALSE)
+		if(!istype(parent,/atom))
+			return COMPONENT_INCOMPATIBLE
+		. = ..()
+		src.reagent_id = reagent_id
+		src.units = units
+		src.overflowing = overflowing
+
+		if(!istext(src.reagent_id) || isnull(reagents_cache[src.reagent_id]))
+			return
+		if(src.units <= 0)
+			return
+
+		if(src.overflowing)
+			RegisterSignal(parent, list(COMSIG_ATOM_EXAMINE), .proc/examined)
+
+		if(isitem(parent))
+			RegisterSignal(parent, list(COMSIG_ITEM_PROCESS), .proc/ticked)
+			if(!(parent in global.processing_items))
+				global.processing_items.Add(parent)
+				src._added_to_items_processing = TRUE
```

Comment:
No, this is a bad way of doing this, it will only lead to issues. Why register the item itself and not the component to some loop?

I hate that `COMSIG_ITEM_PROCESS` exists.

Reply:
comsig_item_process is very dangerous it leads to code like this often iirc rads and cell_holder do it too probably should be fixed